fuel-compression/src/key.rs

```diff
@@ -53,6 +54,26 @@ impl TryFrom<u32> for RegistryKey {
     }
 }
 
+impl TryFrom<&[u8]> for RegistryKey {
+    type Error = &'static str;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        if value.len() != Self::SIZE {
+            return Err("RegistryKey must be 3 bytes long");
+        }
+
+        let mut bytes = [0u8; 3];
+        bytes.copy_from_slice(value);
+        Ok(Self(bytes))
+    }
+}
```

Comment:
Nit: How about using a dedicated error type here?
```suggestion
impl TryFrom<&[u8]> for RegistryKey {
    type Error = InvalidRegistryKeySize;

    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
        if value.len() != Self::SIZE {
            return Err(InvalidRegistryKeySize);
        }

        let mut bytes = [0u8; 3];
        bytes.copy_from_slice(value);
        Ok(Self(bytes))
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, derive_more::Display, derive_more::Error)]
#[display("RegistryKey must be 3 bytes long")]
struct InvalidRegistryKeySize;
```